File: src/main/java/org/nitk/hackathon/bank/TestBankManager.java
package org.nitk.hackathon.bank;

import org.nitk.hackathon.account.Account;

public class TestBankManager {
    private TestBank bank;

    public TestBankManager(TestBank bank) { 
        this.bank = bank;
    }

    public void depositAccount(Account account, double amount) {
        bank.addTransaction(account, amount, new DepositAmount());
    }

    public void withdrawAccount(Account account, double amount) {
        bank.addTransaction(account, amount, new WithdrawAmount());
    }

    public void getBalance(Account account) {
        bank.getBalance(account);
    }
}

File: python/bank/test_bank_manager.py
from python.account.account import Account
from python.bank.deposit_amount import DepositAmount
from python.bank.test_bank import TestBank
from python.bank.withdraw_amount import WithdrawAmount

class TestBankManager:
    """
    A class representing a TestBankManager.
    """

    def __init__(self, bank: TestBank):
        """
        Initialize the TestBankManager object with a TestBank instance.
        """
        self.bank = bank

    def deposit_account(self, account: Account, amount: float):
        """
        Deposit an amount into an account.
        """
        self.bank.add_transaction(account, amount, DepositAmount())

    def withdraw_account(self, account: Account, amount: float):
        """
        Withdraw an amount from an account.
        """
        self.bank.add_transaction(account, amount, WithdrawAmount())

    def get_balance(self, account: Account):
        """
        Get the balance of an account.
        """
        self.bank.get_balance(account)
